src/components/vertical-menu/vertical-menu.pw.tsx

```diff
@@ -0,0 +1,702 @@
+    page,
+  }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Enter");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await expect(
+      verticalMenuItem(page).nth(4).filter({ hasText: "Active item" })
+    ).toBeFocused();
+  });
+
+  test(`should check the focus styling, focusRedesignOptOut true`, async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(<VerticalMenuDefaultComponent />, {
+      hooksConfig: { focusRedesignOptOut: true },
+    });
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    await checkGoldenOutline(verticalMenuItem(page).nth(1));
+  });
+
+  test(`should check the focus styling, focusRedesignOptOut false`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<VerticalMenuDefaultComponent />);
+
+    await verticalMenuItem(page).first().focus();
+    await page.keyboard.press("Tab");
+    const menuItem2 = verticalMenuItem(page).nth(1);
+    await expect(menuItem2).toHaveCSS(
+      "box-shadow",
+      "rgba(0, 0, 0, 0.9) 0px 0px 0px 3px inset, rgb(255, 188, 25) 0px 0px 0px 6px inset"
+    );
+    await expect(menuItem2).toHaveCSS("outline", "rgba(0, 0, 0, 0) solid 3px");
+  });
+});
+
+test.describe(
+  "VerticalMenuFullScreen test background scroll when tabbing",
+  () => {
+    test(`tabbing forward through the menu and back to the start should not make the background scroll to the bottom`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<VerticalMenuFullScreenBackgroundScrollTest />);
+
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await expect(closeIconButton(page)).toBeFocused();
+      await expect(page.getByTestId("#bottom-box")).not.toBeInViewport();
+    });
+
+    test(`tabbing backward through the menu and back to the start should not make the background scroll to the bottom`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<VerticalMenuFullScreenBackgroundScrollTest />);
+
+      await page.keyboard.press("Shift+Tab");
+      await page.keyboard.press("Shift+Tab");
+      await page.keyboard.press("Shift+Tab");
+      await expect(closeIconButton(page)).toBeFocused();
+      await expect(page.getByTestId("#bottom-box")).not.toBeInViewport();
+    });
+  }
+);
+
+test.describe("Events test", () => {
+  test(`should call onClick callback when a click event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <VerticalMenuTriggerCustom
+        onClick={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await verticalMenuTrigger(page).click();
+
+    expect(callbackCount).toBe(1);
```

Comment:
suggestion/question(non-blocking): same for the others too. I thought you'd need an `await` before this?
```suggestion
    await expect(callbackCount).toBe(1);
```